doc/modules/linear_model.rst

```diff
@@ -860,28 +860,69 @@ regularization.
-regression minimizes the following cost function:
+Binary Case
+-----------
 
-.. math:: \min_{w, c} \frac{1}{2}w^T w + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1) .
+For notational ease, we assume that the target :math:`y_i` takes values in the
+set :math:`\{0, 1\}` for data point :math:`i`.
 
-Similarly, :math:`\ell_1` regularized logistic regression solves the following
-optimization problem:
+Once fitted, the ``predict_proba`` method of ``LogisticRegression`` predicts
+the class probability of
 
-.. math:: \min_{w, c} \|w\|_1 + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1).
+.. math:: P(y_i=1|X_i) = \operatorname{expit}(X_i^T w + w_0) = \frac{1}{1 + \exp(-X_i^T w - w_0)}.
 
-Elastic-Net regularization is a combination of :math:`\ell_1` and
-:math:`\ell_2`, and minimizes the following cost function:
+As an optimization problem, binary
+class logistic regression with regularization term :math:`r(w)`  minimizes the
+following cost function:
 
-.. math:: \min_{w, c} \frac{1 - \rho}{2}w^T w + \rho \|w\|_1 + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1),
+.. math:: \min_{w, c} r(w) + C \sum_{i=1}^n \left[-y_i P(y_i=1|X_i)) - (1 - y_i) \log(1 - P(y_i=1|X_i))\right].
 
-where :math:`\rho` controls the strength of :math:`\ell_1` regularization vs.
-:math:`\ell_2` regularization (it corresponds to the `l1_ratio` parameter).
+Multinomial Case
 
-Note that, in this notation, it's assumed that the target :math:`y_i` takes
-values in the set :math:`{-1, 1}` at trial :math:`i`. We can also see that
-Elastic-Net is equivalent to :math:`\ell_1` when :math:`\rho = 1` and equivalent
-to :math:`\ell_2` when :math:`\rho=0`.
+We currently implement four choices of regularization term
+
+#. None, :math:`r(w) = 0`
+#. :math:`\ell_1,\, r(w) = \|w\|_1`
+#. :math:`\ell_2,\, r(w) = \frac{1}{2}\|w\|_2^2 = \frac{1}{2}w^T w`
+#. ElasticNet, :math:`r(w) = \frac{1 - \rho}{2}w^T w + \rho \|w\|_1`
+
+For ElasticNet, :math:`\rho` (which corresponds to the `l1_ratio` parameter)
+controls the strength of :math:`\ell_1` regularization vs. :math:`\ell_2`
+regularization. Elastic-Net is equivalent to :math:`\ell_1` when
+:math:`\rho = 1` and equivalent to :math:`\ell_2` when :math:`\rho=0`.
+
+----------------
+
+The binary case can be extended to :math:`K`-classes leading to the multinomial logistic regression, see also `log-linear model
+<https://en.wikipedia.org/wiki/Multinomial_logistic_regression#As_a_log-linear_model>`_.
+
+.. note::
+   It is possible in a :math:`K`-class context to parameterize the model
+   using only :math:`K-1` weight vectors, leaving one class probability fully
+   determined by the other class probabilities by leveraging the fact that all
+   class probabilities must sum to one. We deliberately choose to overparameterize the model
+   using :math:`K` weight vectors for ease of implementation and to preserve the
+   symmetrical inductive bias regarding ordering of classes, see [1].. This effect becomes
```

Comment:
```suggestion
   symmetrical inductive bias regarding ordering of classes, see [1]. This effect becomes
```
The reference [1] is missing. And it must get a different number (5). Then, all following references must be incremented.